src/Cowsay.zig

```diff
@@ -11,23 +11,17 @@ const default_cow =
     \\    ||     ||
 ;
 
-var general_purpose_allocator = std.heap.GeneralPurposeAllocator(.{}){};
-
-
-
 /// writer for output. must be an AnyWriter. Use `.any()` to convert to AnyWriter before passing in.
-w: std.io.AnyWriter,
+writer: std.io.AnyWriter,
 /// the eyes. will substitute the first two `o` of the cow
 eyes: [2]u8 = [2]u8{ 'o', 'o' },
 
-/// allocator to use when formatting message
-allocator: std.mem.Allocator = general_purpose_allocator.allocator(),
 /// private variables
 thinking: bool = false,
-max_line_length: u32 = 0,
-offset: u32 = 0,
+max_line_length: usize = 0,
+offset: usize = 0,
+cow: []u8 = &.{},
```

Comment:
Instead of manually keeping track of buffer length, instead use a slice that points to the buffer! This not only keeps track of the length for you, but also allows for convenient slicing/indexing operations, being directly passed around to various standard library functions, and is generally nicer to work with.